consensus/bitcoin-ng.md

```diff
@@ -84,9 +84,41 @@ Bitcoin-NG inherits Bitcoin's ability to fork in case of multiple block producer
         mempool content MB(1)' and MB(2) should be very similar. Re-signing
         the delta of transactions missing in Leader 2's node is very cheap operation.
 
+Proof of Fraud
+==
 
-
-References:
+A malicious leader could construct forks in a generation of micro
+blocks, either by forking directly from his key block, or by forking
+on a micro block. This could be done to disrupt the network or to
+perform double spend attacks.
+
+To prevent this malicious behaviour, the miner of the next key block
+can submit Proof-of-Fraud in a micro block in its generation. Only
+this leader can submit Proof-of-Fraud to avoid flooding the network
+with reports, which in itself could be viewed as a DoS
+attack. Furthermore, reporting a key block more than once is regarded
+as a protocol violation, even if there is more than one act of fraud
+committed.
+
+The Proof-of-Fraud is constructed of two sibling micro headers (with
+the same previous pointer) along with the public key of the
+signer. The Proof-of-Fraud is included in a micro block body, and the
+header must contain the hash of the Proof-of-Fraud object to mark its
+existence in the block.
```

Comment:
So there is no constraint that at least one of the two reported microblock has been included in the chain. That makes sense. Ok.